central/sensor/augmentation/broker.go

```diff
@@ -0,0 +1,88 @@
+package augmentation
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// The Broker coordinates and matches deployment enhancement requests to responses
+type Broker struct {
+	requests map[string]chan<- *central.DeploymentEnhancementResponse
+	lock     sync.Mutex
+	once     sync.Once
+}
+
+// NewBroker returns a new broker
+func NewBroker() *Broker {
+	return &Broker{
+		requests: make(map[string]chan<- *central.DeploymentEnhancementResponse),
```

Comment:
I think, I am starting to understand what is going on here. You want to wait for each sensor to send some information. Once a given sensor replies, then some actions can happen. 
I think, that current implementation has some issues - especially there is a problem with closing the channel after the first message, while having no guarantees that there will be exactly one message on the channel.

I think, that much cleaner implementation would be to have a single channel that would be handling messages of type:

```go
type chanMsg struct{
    id   string
    msg  *central.DeploymentEnhancementResponse
}
```

(or just the `DeploymentEnhancementResponse` as it contains the ID already, right?)

This will allow you to keep the channel open for entire lifecycle of the central pipeline. You can handle as many duplicate messages as you like and would make it much simpler.

If I am misunderstanding your use case then please let me know (it may be good to meet an chat about that).